app/src/main/java/openfoodfacts/github/scrachx/openfood/views/holders/HistoryScanHolder.java

```diff
@@ -42,6 +43,8 @@ public HistoryScanHolder(final View itemView, final String productUrl, Activity
             if (isConnected) {
                 OpenFoodAPIClient api = new OpenFoodAPIClient(mActivity);
                 api.getProduct(txtBarcode.getText().toString(), (Activity) v.getContext());
+            } else {
+                Toast.makeText(mActivity, "Sorry! There was a network error! Please check your internet connection", Toast.LENGTH_SHORT).show();
```

Comment:
Create a new resource string instead of hardcoding it so it can be translated.
[See here](https://developer.android.com/guide/topics/resources/string-resource.html) for documentation.

Reply:
@Vai Ton I have made the changes and squashed the commits. Please share if you have any other suggestion as well:)

Reply:
"The product couldn't be loaded. Please try again when you are back online." 
The popup is just hightlighting a weakness of the app (ie we should store a copy of the history for offline viewing)

Reply:
@teolemon I have made the changes. Kindly review:)